#[jy issue 36] express와 gmail api 연동했습니다.

jaeyeon93 opened the pull request:
## Related Issues
resolve: [36]

## Description
- api라우터를 생성했습니다.
- apiService를 만들어서 여기서 base64toUtf8, getMessage, getList 등 이메일을 가져오는 메서드들을 구현했습니다.

jaeyeon93 committed:
#36 [gmail api로 list요청] router로 api/list를 하면 mail의 id를 가져온다.

- server/env디렉토리를 추가. token, credentials를 관리하므로 gitignore에 추가했다.
- apiService라는 컴포넌트를 생성. gmail api의 기본 코드들과 getList, getMessage 메서드 추가
- routes/api.js 라우터추가. 기본 라우터 api/...

jaeyeon93 committed:
#36 [gmail api로 list가져오기, 내용가져오기] 메일 목록, 개별메일 가져오기

- GET /api/list로 Apple이 보낸 이메일만 가져오기.
- GET /api/:id로 이메일의 내용을 출력한다.
- base64toUtf8로 converting

parkyounghwan commented on server/components/api/apiService.js:
```diff
@@ -0,0 +1,84 @@
+  console.log('getList 호출');
+  const gmail = google.gmail({version: 'v1', auth});
+  return new Promise((resolve, reject) => {
+    gmail.users.messages.list({userId: 'jimmyjaeyeon@gmail.com'}, (err, res) => {
```
개발 중이기는 하지만, 이메일 처럼 안 변하는 정보들 상수 처리 위에 'CREDENTIALS_PATH' 이런 것 처럼

parkyounghwan commented:
예스 api 디렉토리 따로 빼 주^3^

jaeyeon93 committed:
#36 [PR 피드백 수정] 코드리뷰 피드백 수정

- api디렉토리 따로 분리
- 고정된 값들 const를 이용한 상수값으로 분리

jaeyeon93 committed:
#36 [PR 피드백 수정] api디렉토리를 server디렉토리에서 분리

- npm init으로 package.json추가
- package.json에 googleapis, express모듈 추가 후 설치
- env디렉토리 api의 하위디렉토리로 이동

jaeyeon93 committed:
#36 [.gitignore] gitignore파일수정

- node_modules, package-lock.json파일 gitignore에 추가

jaeyeon93 committed:
#36 [git push 오류 해결중] package.json추가

- api/package.json추가
